src/libstd/path.rs

```diff
@@ -1397,6 +1398,36 @@ impl From<String> for PathBuf {
     }
 }
 
+/// An error returned when parsing `Path` or `PathBuf` using `from_str` fails.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[stable(feature = "rust1", since = "1.0.0")]
+pub struct ParsePathError { _priv: () }
```

Comment:
Since parsing a Path will never fail, please make it an empty enum:

```rust
pub enum ParsePathError {}
```

Alternatively, simply return `std::string::ParseError`.

Reply:
We shouldn't reuse an existing type.